### PR TITLE
[RHCLOUD-30750] Moving to platform-middlware-go v2 package for identity and logging

### PR DESCRIPTION
**Pull request by tahmidefaz**
## What?
Explain what the change is linking any relevant JIRAs or Issues.

When moving to `platform-middlware-go v2`, changes were needed in `identity management`, `request_id fetching`, and `cloudwatch logging`.

## Why?
Consider what business or engineering goal does this PR achieves.

## How?
Describe how the change is implemented. Any noteable new libaries, APIs, or features.

## Testing
Did you add any tests for the change?

## Anything Else?
Any other notes about the PR that would be useful for the reviewer. 

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by gmcculloug** [comment]
Excited to see this PR.  Thanks @tahmidefaz 🙌

**Review by dehort — CHANGES_REQUESTED** [review]
The changes look good to me, but it looks like the build is failing.

I can try to take a closer look at the failures later.

**Comment by tahmidefaz** [comment]
> The changes look good to me, but it looks like the build is failing.
> 
> I can try to take a closer look at the failures later.

I was just looking into it, no error message. It just fails. :shrug: 
Maybe the build pipeline needs to be updated, not sure.

File: unit_test.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 export GO111MODULE="on"
 export GOPATH="$(pwd)/gopath"
-export GOROOT="/opt/go/1.19.11"
+export GOROOT="/opt/go/1.21.3"
+echo | ls /opt/go
```
Comment (skarekrow) [inline unit_test.sh:7]: Don't forget to remove the debug :D

**Pull request merged**